core/src/main/java/com/yahoo/oak/InternalOakBasics.java

```diff
@@ -15,22 +15,30 @@
     /*-------------- Members --------------*/
     protected static final int MAX_RETRIES = 1024;
 
-    private final MemoryManager valuesMemoryManager;
-    private final MemoryManager keysMemoryManager;
+    protected final OakSharedConfig<K, V> config;
 
-    private final OakSerializer<K> keySerializer;
-    private final OakSerializer<V> valueSerializer;
+    protected final MemoryManager keysMemoryManager;
```

Comment:
Why to keep the config and all those fields? In order to avoid de-referencing? I don't believe we will see any performance differences from accessing `config.keysMemoryManager`...  We may try the benchmarks....

Reply:
Comparators are accessed frequently. Are you sure?
I think we can have a different PR after this one where we test this hypothesis.